compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -977,6 +977,21 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         can_suggest_clone
     }
 
+    pub(crate) fn clone_on_reference(&self, expr: &hir::Expr<'_>) -> Option<Span> {
```

Comment:
please add a doc comment, I don't fully understand this function.

Reply:
it's poinit is to detect `x.clone()` calls where the type of `x` matches the return type which is caused by an auto-ref?

Reply:
Yes, correct.